pypsa/plot.py

```diff
@@ -254,15 +253,95 @@ def as_branch_series(ser):
+    xy1 = minxy - margin*(maxxy - minxy)
+    xy2 = maxxy + margin*(maxxy - minxy)
+    return tuple(xy1), tuple(xy2)
+
+
+def projected_area_factor(ax):
+    """
+    Helper function to get the area scale of the current projection in
+    reference to the default projection.
+    """
+    if not 'projection' in ax.__dict__:
+        return 1
+    if ax.projection is ccrs.PlateCarree():
+        return 1
+    x1, x2, y1, y2 = ax.get_extent()
+    pbounds = \
+        ccrs.PlateCarree().transform_points(ax.projection,
+                    np.array([x1, x2]), np.array([y1, y2]))
+
+    return np.sqrt(abs((x2 - x1) * (y2 - y1))
+                   /abs((pbounds[0] - pbounds[1])[:2].prod()))
+
+
+
+def draw_map(network=None, jitter=None, ax=None, boundaries=None,
+             margin=0.05, drawmap=True):
+
+    x, y = network.buses["x"],  network.buses["y"]
+
+    if jitter is not None:
+        x = x + np.random.uniform(low=-jitter, high=jitter, size=len(x))
+        y = y + np.random.uniform(low=-jitter, high=jitter, size=len(y))
+
+    if boundaries is None:
+        (x1, y1), (x2, y2) = compute_bbox_with_margins(margin, x, y)
+    else:
+        x1, x2, y1, y2 = boundaries
+
+    #First choice should be cartopy
+    if cartopy_present:
+        resolution = '50m' if isinstance(drawmap, bool) else drawmap
+        assert resolution in ['10m', '50m', '110m'], (
+                "Resolution has to be one of '10m', '50m', '110m'")
+        geomap = ax.projection
+        transformed = pd.DataFrame(
+                    geomap.transform_points(
+                        ccrs.PlateCarree(), x.values, y.values),
+                        columns=['x', 'y', 'z'], index=network.buses.index)
+        x, y = transformed.x, transformed.y
+        ax.set_extent([x1, x2, y1, y2], crs=ccrs.PlateCarree())
+        ax.coastlines(linewidth=0.4, zorder=-1, resolution=resolution)
+        border = cartopy.feature.BORDERS.with_scale(resolution)
+        ax.add_feature(border, linewidth=0.3)
+
+    elif basemap_present:
+        resolution = 'l' if isinstance(drawmap, bool) else drawmap
+        geomap = Basemap(resolution=resolution, epsg=network.srid,
+                       llcrnrlat=y1, urcrnrlat=y2, llcrnrlon=x1,
+                       urcrnrlon=x2, ax=ax)
+        geomap.drawcountries(linewidth=0.3, zorder=-1)
+        geomap.drawcoastlines(linewidth=0.4, zorder=-1)
+
+        x, y = geomap(x.values, y.values)
+        x = pd.Series(x, network.buses.index)
+        y = pd.Series(y, network.buses.index)
+
+    return geomap, x, y
+
+
 #This function was borne out of a breakout group at the October 2017
 #Munich Open Energy Modelling Initiative Workshop to hack together a
 #working example of plotly for networks, see:
-#https://forum.openmod-initiative.org/t/breakout-group-on-visualising-networks-with-plotly/384/7
+#https://forum.openmod-initiative.org/t/
+#breakout-group-on-visualising-networks-with-plotly/384/7
```

Comment:
I'd prefer not to break the url, so you can copy it easily. Also use spaces after the hash if possible.